fs/metrics/common/metrics.go

```diff
@@ -62,22 +62,61 @@ const (
 
 // Lists all metric labels.
 const (
-	// prometheus metrics
-	Mount                         = "mount"
-	RemoteRegistryGet             = "remote_registry_get"
-	NodeReaddir                   = "node_readdir"
-	InitMetadataStore             = "init_metadata_store"
-	ReadOnDemand                  = "read_on_demand"
-	MountLayerToLastOnDemandFetch = "mount_layer_to_last_on_demand_fetch"
+	/**
+	Metrics emitted via prometheus endpoint. All latency metrics are measured in milliseconds,
+	unless specified otherwise.
+	*/
+	// Mount measures the duration of Mount operation for resolved layer.
+	Mount = "mount"
+
+	// RemoteRegistryGet records the roundtrip latency for remote registry GET operation .
+	RemoteRegistryGet = "remote_registry_get"
+
+	// NodeReaddir measures how long readdir operation takes (in microseconds).
+	NodeReaddir = "node_readdir"
+
+	// InitMetadataStore measures how long it takes to initialize the file metadata.
+	InitMetadataStore = "init_metadata_store"
+
+	// ReadOnDemand measures the latency of on-demand reads (in mocroseconds).
+	ReadOnDemand = "read_on_demand"
 
-	OnDemandReadAccessCount          = "on_demand_read_access_count"
+	// OnDemandReadAccessCount records the count of on-demand reads.
+	OnDemandReadAccessCount = "on_demand_read_access_count"
+
+	// OnDemandRemoteRegistryFetchCount records the count of the on-demand fetch requests to remote registry.
 	OnDemandRemoteRegistryFetchCount = "on_demand_remote_registry_fetch_count"
-	OnDemandBytesServed              = "on_demand_bytes_served"
-	OnDemandBytesFetched             = "on_demand_bytes_fetched"
 
-	// logs metrics
-	BackgroundFetchTotal      = "background_fetch_total"
-	BackgroundFetchDownload   = "background_fetch_download"
+	// OnDemandBytesServed counts the number of bytes served on-demand.
+	OnDemandBytesServed = "on_demand_bytes_served"
+
+	// OnDemandBytesFetched counts the number of bytes fetched on-demand.
+	OnDemandBytesFetched = "on_demand_bytes_fetched"
+
+	/**
+	Metrics logged in snapshotter's log.
+	All latencies here are measured in milliseconds, unless stated otherwise.
+	*/
+	// MountLayerToLastOnDemandFetch measures the latency of last on demand fetch,
+	// which must be invoked at the end of background fetch operation only.
```

Comment:
I'm not following here. Why does MountLayerToLastOnDemandFetch need to be invoked at the end of _background_ fetch?

Reply:
This metric is inherited from stargz snapshotter. The idea is that on-demand fetches finish when background fetch is complete, since all the data for the layer is local. I'm just documenting it as it is right now. Most likely, we need to revisit the approach, given that we plan to do changes in background fetcher.

Reply:
Do we log this metric now?

Reply:
Yes, it's logged here: https://github.com/awslabs/soci-snapshotter/blob/main/fs/fs.go#L561

Reply:
We inherited this metric from stargz snapshotter. Renaming and reworking that metric should be a topic of the other PR after we conclude the broader conversation on the metrics. This PR just documents what is already in there. 
On that topic, probably, it will make sense to emit it with via the prometheus endpoint, so we capture the time when last on-demand fetch for that particular layer was done.

Reply:
ok, I can approve this PR, but I still think we should rename it in this PR because the current name is confusing and does not reflect what is being logged, and it won't take much effort to change a name.